backend/src/users/users.controller.ts

```diff
@@ -68,4 +86,50 @@ export class UsersController {
   remove(@Param('id') id: string): Promise<DeleteResult> {
     return this.usersService.remove(+id);
   }
+
+  @Post(':id/avatar')
+  @UseInterceptors(FileInterceptor('file'))
+  @ApiBearerAuth()
+  @ApiResponse({ status: 201, description: 'Success' })
+  @ApiConsumes('multipart/form-data')
+  @ApiBody({
+    description: 'Avatar Photo',
+    type: AvatarUploadDto,
+  })
+  uploadAvatar(
+    @Param('id') id: string,
+    @UploadedFile() file: Express.Multer.File,
+  ) {
+    return this.usersService.updateAvatar(
+      +id,
+      `${this.configService.get<string>('UPLOADS_PATH')}` + file.filename,
+    );
+  }
+
+  @Get(':id/avatar')
+  @ApiBearerAuth()
+  @ApiResponse({ status: 200, description: 'Success' })
+  @ApiResponse({ status: 204, description: 'No avatar found' })
+  @ApiResponse({ status: 400, description: 'User does not exist' })
+  async getAvatar(
+    @Param('id') id: string,
+    @Response({ passthrough: true }) res,
+  ): Promise<StreamableFile> {
+    try {
+      const file = await this.usersService.getAvatar(+id);
+      res.set({
+        'Content-Type': `image/${file.ext}`,
+        'Content-Disposition': `inline; filename="avatar.${file.ext}"`,
+      });
+      return file.fileStream;
+    } catch (error) {
+      if (error instanceof EntityDoesNotExistError) {
+        if (error.message === 'Avatar does not exist') {
+          throw new HttpException(error.message, HttpStatus.NO_CONTENT);
+        }
+        throw new BadRequestException(error.message);
+      }
+      throw error;
```

Comment:
I am not sure when this happens. The correction asks us never to send a 5xx. Maybe wrap it in a 418?

Reply:
It shouldn't happen, it was just a catchall in case an unforeseen error happened (problems writing to disk, etc). But we can remove it. In most cases it will throw a `NO_CONTENT` exception, unless it is a bad request from the client